src/lib/Form/Type/ContentType/ContentTypeChoiceType.php

```diff
@@ -39,15 +46,7 @@ public function configureOptions(OptionsResolver $resolver)
     {
         $resolver
             ->setDefaults([
-                'choice_loader' => new CallbackChoiceLoader(function () {
-                    $contentTypes = [];
-                    $contentTypeGroups = $this->contentTypeService->loadContentTypeGroups();
-                    foreach ($contentTypeGroups as $contentTypeGroup) {
-                        $contentTypes[$contentTypeGroup->identifier] = $this->contentTypeService->loadContentTypes($contentTypeGroup);
-                    }
-
-                    return $contentTypes;
-                }),
+                'choice_loader' => $this->contentTypeChoiceLoader,
```

Comment:
Does it need to be registered as a service? I'm pretty sure in Symfony's codebase they are initializing ChoiceLoaders on demand which for me makes sense as ChoiceLoader living in the container wastes memory. Large number of services makes your application slower.